source/extensions/stat_sinks/common/statsd/statsd.cc

```diff
@@ -330,7 +330,7 @@ void TcpStatsdSink::TlsSink::write(Buffer::Instance& buffer) {
   //       since if we stay over, the other threads will eventually kill their connections too.
   // TODO(mattklein123): The use of the stat is somewhat of a hack, and should be replaced with
   // real flow control callbacks once they are available.
-  if (parent_.cluster_info_->trafficStats().upstream_cx_tx_bytes_buffered_.value() >
+  if (parent_.cluster_info_->trafficStats()->upstream_cx_tx_bytes_buffered_.value() >
```

Comment:
Put the instantiated traffic stats in a local ref since you are going to access them a bunch of times. Why do all those extra atomics and conditionals?

In general I think it would help to think more about when in a request flow we can instantiate the stats and then use them rather than doing to the lazy-load on every inc. But some of that reasoning may be hard. In this case the reasoning is easy.